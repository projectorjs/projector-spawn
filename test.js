// @flow
'use strict';

const spawn = require('./');
const path = require('path');

const fixture = path.join(__dirname, 'fixture.js');

function expectError() {
  throw new Error('Promise should have errored instead of fulfilling.');
}

test('success', () => {
  return spawn(fixture, ['SUCCESS']).then(res => {
    expect(res).toMatchSnapshot();
  });
});

test('failure', () => {
  return spawn(fixture, ['FAILURE']).then(expectError, err => {
    expect(err.stack).toMatchSnapshot('failure err.stack');
    expect(err.code).toMatchSnapshot('failure err.code');
    expect(err.stdout).toMatchSnapshot('failure err.stdout');
    expect(err.stderr).toMatchSnapshot('failure err.stderr');
  });
});

test('throw', () => {
  return spawn(fixture, ['THROW']).then(expectError, err => {
    expect(err.stack).toMatchSnapshot('throw err.stack');
    expect(err.code).toMatchSnapshot('throw err.code');
    expect(err.stdout).toMatchSnapshot('throw err.stdout');
    expect(err.stderr).toMatchSnapshot('throw err.stderr');
  });
});
