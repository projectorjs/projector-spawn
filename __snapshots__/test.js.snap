// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`failure err.code 1`] = `1`;

exports[`failure err.stack 1`] = `
"Error: what have you done
    at ChildProcess.child.on.code (/Users/jkyle/Projects/projector-spawn/index.js:53:16)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:897:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)"
`;

exports[`failure err.stderr 1`] = `"what have you done"`;

exports[`failure err.stdout 1`] = `"argv: [\\"/usr/local/Cellar/node/8.1.3/bin/node\\",\\"/Users/jkyle/Projects/projector-spawn/fixture.js\\",\\"FAILURE\\"]"`;

exports[`success 1`] = `
Object {
  "code": 0,
  "stderr": "",
  "stdout": "argv: [\\"/usr/local/Cellar/node/8.1.3/bin/node\\",\\"/Users/jkyle/Projects/projector-spawn/fixture.js\\",\\"SUCCESS\\"]",
}
`;

exports[`throw err.code 1`] = `1`;

exports[`throw err.stack 1`] = `
"Error: /Users/jkyle/Projects/projector-spawn/fixture.js:15
  throw new Error('no');
  ^

Error: no
    at Object.<anonymous> (/Users/jkyle/Projects/projector-spawn/fixture.js:15:9)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3

    at ChildProcess.child.on.code (/Users/jkyle/Projects/projector-spawn/index.js:53:16)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:897:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)"
`;

exports[`throw err.stderr 1`] = `
"/Users/jkyle/Projects/projector-spawn/fixture.js:15
  throw new Error('no');
  ^

Error: no
    at Object.<anonymous> (/Users/jkyle/Projects/projector-spawn/fixture.js:15:9)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
"
`;

exports[`throw err.stdout 1`] = `"argv: [\\"/usr/local/Cellar/node/8.1.3/bin/node\\",\\"/Users/jkyle/Projects/projector-spawn/fixture.js\\",\\"THROW\\"]"`;
